### PR TITLE
Add flags to specify HSTS / Frame-Options / CSP headers for httpserver

### PR DESCRIPTION
**Pull request by dima-vm**
Added the following flags:

* `-http.header.csp` string
     Value for 'Content-Security-Policy' header (default "default-src 'self'")
*  `-http.header.frameOptions` string
     Value for 'X-Frame-Options' header (default "SAMEORIGIN")
* `-http.header.hsts` string
     Value for 'Strict-Transport-Security' header (default "max-age=31536000; includeSubDomains")

### CODE REVIEW

File: lib/httpserver/httpserver.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ var (
 	tlsCipherSuites = flagutil.NewArrayString("tlsCipherSuites", "Optional list of TLS cipher suites for incoming requests over HTTPS if -tls is set. See the list of supported cipher suites at https://pkg.go.dev/crypto/tls#pkg-constants")
 	tlsMinVersion   = flag.String("tlsMinVersion", "", "Optional minimum TLS version to use for incoming requests over HTTPS if -tls is set. "+
 		"Supported values: TLS10, TLS11, TLS12, TLS13")
+	httpHSTS         = flag.String("http.header.hsts", "", "Value for 'Strict-Transport-Security' header, example: 'max-age=31536000; includeSubDomains'")
```
Comment (hagen1778) [inline lib/httpserver/httpserver.go:39]: should we have some default value here? What is the default requirement for such header?

File: lib/httpserver/httpserver.go [hunk 2]
```diff
@@ -36,6 +36,9 @@ var (
...
+	httpFrameOptions = flag.String("http.header.frameOptions", "SAMEORIGIN", "Value for 'X-Frame-Options' header")
+	httpCSP          = flag.String("http.header.csp", "", "Value for 'Content-Security-Policy' header")
```
Comment (hagen1778) [inline lib/httpserver/httpserver.go:41]: Should we have a default policy here?

**Review by hagen1778 — COMMENTED** [review]
@dima-vm I think it is worth to update README with new flag descriptions so they become searchable via google or docs search engine.
I think we should also mention these flags in https://docs.victoriametrics.com/#security
wdyt?

**Comment by dima-vm** [comment]
@hagen1778 Also added new flags to all 21 docs where other httpserver flags are documented.

**Comment by hagen1778** [comment]
Thanks @dima-vm! Please note, I removed default values  for new flags in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5111/commits/d6e2db293251b390a36da6362a4c1a3c9785cf40 as I found the change disruptive for local and testing installations.

**Review by hagen1778 — APPROVED** [review]

**Comment by dima-vm** [comment]
Hi @hagen1778 ,
What exactly errors were there brought in from testing? Maybe we could fix those and keep the more secure defaults? At least for some headers.
Ugh, I see you merged it in already, didn't expect that.

**Comment by hagen1778** [comment]
Hi @dima-vm ! I found that some static assets failed to be loaded for various VM web components. I read docs for all these headers and realized that it might be not straightforward to avoid changes in behavior. For example, forcing `Strict-Transport-Security` is something that could backfire unexpectedly.
I also checked for default values for these headers in Prometheus and found they are all empty by default. So I chose a safe path here.

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.95.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Pull request merged**